File: 27/homework1(dynamic programming)/2/2.py
with open('27B.txt', 'r', encoding='utf-8') as file:
    info = int(file.readline())

    k5 = k13 = k65 = k = 0
    count = 0
    
    for i in range(info):
        num = int(file.readline())
        # Здесь получается инь и ян
        if num % 65 == 0: count += k
        elif num % 5 == 0: count += k13
        elif num % 13 == 0: count += k5
        else: count += k65

        # Здесь вообще песня, учитываем ВСЕ!!!!!!
        k += 1
        if num % 65 == 0: k65 += 1
        if num % 13 == 0: k13 += 1 
        if num % 5 == 0: k5 += 1
    print(count)

#  1) 168
#  2) 2503584

File: 27/homework(subsequences)/2/2.cpp
#include <iostream>
#include <vector>
#include <fstream>
using namespace std;

int main() {
	int n;
	fstream f("27B.txt");
	f >> n;
	
	vector <int> l(n);
	for(int i = 0; i < n; i++) f >> l[i];
	
	long long count = 0;
	
	for(int i = 0; i < n; i ++) {
		long long s = 0;
		for(int j = i; j < n; j++) {
			s += l[j];
			if (s % 666 == 0) count ++;
			
		}
	}
	cout << count;
	
	return 0;
}
